src/app/pages/Sandbox/Editor/Workspace/SandboxActions/index.js

```diff
@@ -28,18 +24,43 @@ const PrivacySelect = styled.select`
   box-sizing: border-box;
 `;
 
-export default class SandboxSettings extends React.PureComponent {
+const mapDispatchToProps = dispatch => ({
+  modalActions: bindActionCreators(modalActionCreators, dispatch)
+});
+
+type Props = {
+  id: string,
+  deleteSandbox: (id: string) => void,
+  newSandboxUrl: () => void,
+  setSandboxPrivacy: (id: string, privacy: number) => void,
+  isPatron: boolean,
+  privacy: 0 | 1 | 2,
+  modalActions: typeof modalActionCreators
+};
+
+class SandboxSettings extends React.PureComponent {
   props: Props;
   state = {
-    loading: false,
+    loading: false
   };
 
-  handleDeleteSandbox = async () => {
-    const really = confirm('Are you sure you want to delete this sandbox?'); // TODO: confirm???
-    if (really) {
-      await this.props.deleteSandbox(this.props.id);
-      await this.props.newSandboxUrl();
-    }
+  handleDeleteSandbox = () => {
+    const { modalActions } = this.props;
+
+    modalActions.openModal({
+      Body: (
+        <Alert
+          title="Delete File"
+          body={<span>Are you sure you want to delete this sandbox?</span>}
+          onCancel={modalActions.closeModal}
+          onDelete={async () => {
+            await this.props.deleteSandbox(this.props.id);
+            await this.props.newSandboxUrl();
```

Comment:
Do these function calls really need to `await`? Right now the modal wouldn't be closed until `deleteSandbox()` and `newSandboxUrl()` are completed. - not sure if that is intended...

Reply:
Hmm, yes, seems like we can remove the `await` statementes.